src/yall.mjs

```diff
@@ -136,6 +137,12 @@ export default function (options) {
 
     if (observeChanges) {
       yallApplyFn(queryDOM(options.observeRootSelector || "body"), yallCreateMutationObserver);
+    } else {
+    // IntersectionObserver not supported
+      if (noPolyfill) {
+        yallApplyFn(lazyElements, yallBindEvents);
+        yallApplyFn(lazyElements, yallLoad);
+      }
```

Comment:
Are you sure you wanted to put this condition here? I think it should go as an `else if` after the `IntersectionObserver` support check, like so:

```javascript
if (io in win && `${io}Entry` in win) {
  var intersectionListener = new win[io](entries => {
    yallApplyFn(entries, yallIntersectionObserve);
  }, {
    rootMargin: `${"threshold" in options ? options.threshold : 200}px 0%`
  });

  yallApplyFn(lazyElements, yallBindEvents);
  yallApplyFn(lazyElements, intersectionListener);

  if (observeChanges) {
    yallApplyFn(queryDOM(options.observeRootSelector || "body"), yallCreateMutationObserver);
  }
} else if (noPolyfill) {
  if (noPolyfill) {
    yallApplyFn(lazyElements, yallBindEvents);
    yallApplyFn(lazyElements, yallLoad);
  }
}
``` 

Otherwise, the `noPolyfill` check ends up taking effect only if `IntersectionObserver` is supported, and if the user doesn't want to use a `MutationObserver`, which I think doesn't accomplish what we're looking to do.

Reply:
You are right. I don't know why I put it here. Fix incoming.